spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -407,7 +413,11 @@ public final boolean send(Message<?> message, long timeout) {
 			if (this.datatypes.length > 0) {
 				message = this.convertPayloadIfNecessary(message);
 			}
-			if (interceptors.getInterceptors().size() > 0) {
+			if (logger.isDebugEnabled()) {
+				logger.debug("preSend on channel '" + this + "', message: " + message);
+			}
+			boolean hasInterceptors = interceptors.getInterceptors().size() > 0;
```

Comment:
Don't see reason in this local var. The `interceptorStack != null` plays the same role.
From other side: shouldn't we apply the same optimization for the `AbstractPollableChannel.receive`?
As well as for the `PollableAmqpChannel` and `PollableJmsChannel` ?

BTW, shouldn't we recommend the same optimization to @rstoyanchev for his `org.springframework.messaging.support.AbstractMessageChannel` ?

As you noticed the `interceptors` may be changed during the `send/receive` invocation and I think you are right that we must not apply that changes to the current invocation.